graphqlapi/src/main/java/io/stargate/graphql/schema/cqlfirst/dml/fetchers/aggregations/AggregationsFetcherSupport.java

```diff
@@ -0,0 +1,222 @@
+            break;
+          case VARINT_FUNCTION:
+            putResultValue(columns, row, selectedField, Row::getBigInteger);
+            break;
+          case FLOAT_FUNCTION:
+            putResultValue(columns, row, selectedField, Row::getFloat);
+            break;
+          case SMALLINT_FUNCTION:
+            putResultValue(columns, row, selectedField, Row::getShort);
+            break;
+          case TINYINT_FUNCTION:
+            putResultValue(columns, row, selectedField, Row::getByte);
+            break;
+        }
+      }
+    }
+    return columns;
+  }
+
+  private void putResultValue(
+      Map<String, Object> columns,
+      Row row,
+      SelectedField selectedField,
+      BiFunction<Row, String, Object> rowValueExtractor) {
+    String alias = selectedField.getAlias();
+    // put the returned value as alias
+    if (alias != null) {
+      columns.put(selectedField.getName(), rowValueExtractor.apply(row, alias));
+    }
+    // generate aggregation column name and get the value
+    else {
+      String columnName = generateAggregationColumnName(selectedField);
+      columns.put(selectedField.getName(), rowValueExtractor.apply(row, columnName));
+    }
+  }
+
+  // it returns C* generated aggregation column name
+  // for example, for the name = count and arguments = ['a']
+  // it will return: 'system.count(a)'
+  private String generateAggregationColumnName(SelectedField selectedField) {
+    Map<String, Object> arguments = selectedField.getArguments();
+    SupportedAggregationFunction functionName = getNameArgument(arguments);
+    List<String> args = getAndValidateArgs(arguments, functionName);
+    return String.format("system.%s(%s)", functionName.getName(), args.get(0));
+  }
+
+  private void addAggregation(
+      Map<String, Object> arguments,
+      QueryBuilder.QueryBuilder__20 queryBuilder,
+      BiFunction<QueryBuilder.QueryBuilder__20, String, QueryBuilder.QueryBuilder__19>
+          addAggregation,
+      SelectedField selectedField,
+      SupportedAggregationFunction supportedAggregationFunction) {
+    List<String> args = getAndValidateArgs(arguments, supportedAggregationFunction);
+    String column = getAndValidateColumn(args, supportedAggregationFunction);
+
+    String alias = selectedField.getAlias();
+    QueryBuilder.QueryBuilder__19 withAggregation = addAggregation.apply(queryBuilder, column);
+    if (alias != null) {
+      withAggregation.as(alias);
+    }
+  }
+
+  private String getAndValidateColumn(
+      List<String> args, SupportedAggregationFunction supportedAggregationFunction) {
+    String column = dbColumnGetter.getDBColumnName(table, args.get(0));
+    if (column == null) {
+      throw new IllegalArgumentException(
+          String.format(
+              "The column name: %s provided for the %s function does not exists.",
+              args.get(0), supportedAggregationFunction.getName()));
+    }
```

Comment:
We have a problem with case-sensitive columns here. For example, given the table:
```
CREATE TABLE foo(k int, cc int, "V" int, PRIMARY KEY (k, cc));
```
There is currently no way to write the GraphQL equivalent of:
```
SELECT count("V") FROM foo WHERE k=1 ;
```
If you pass the column name unquoted in the GraphQL query (as in `_bigint_function(name: "count", args: "V")`), Stargate tries to execute the invalid CQL `SELECT count(v)`.
If you try to quote the column argument (as in `_bigint_function(name: "count", args: "\"V\"")`), then this check fails.

I think the best approach is to expect the first form (exact case, unquoted). That way it's consistent with the DML API, and also the less surprising since it's exactly the same as the GraphQL field name.
We can add the quotes before we pass the argument to the query builder (no need to check if the name is mixed-case or not, `COUNT(v)` and `COUNT("v")` are equivalent, so we can just quote all the time).

Column names are not quoted in CQL result sets, so I _think_ `AggregationFetcherSupport.generateAggregationColumnName` will do the right thing. The best way to check is to add an integration test with case-sensitive names.

Reply:
good catch, fixed in:
https://github.com/stargate/stargate/pull/901/commits/ab21d68bf19b6f3f9a26380b2753830efee827af